src/renderer.js

```diff
@@ -127,19 +127,18 @@ export class Renderer {
   }
 
   constructor(elementId, backend) {
-    this.elementId = elementId;
-    if (!this.elementId) {
+    if (!elementId) {
       throw new RuntimeError('BadArgument', 'Invalid id for renderer.');
     }
 
-    this.element = document.getElementById(elementId);
-    if (!this.element) this.element = elementId;
+    this.element =
+      typeof elementId === 'string' && typeof document !== 'undefined' ? document.getElementById(elementId) : elementId;
```

Comment:
Can document be undefined? When?

Reply:
We can add a console.log of document to see if it ever prints out undefined. When we do the visual diffs via node JS on the command line, is document defined? Are there any other instances where we would be outside of a browser context?

Reply:
Exactly, you can see the commit message, where I have noticed that developers may hope to use this library outside a browser context, like Node, where we can simulate Canvas, and where without document definition.

Reply:
This is fine then.